apps/els_lsp/src/els_sheldon_diagnostics.erl

```diff
@@ -0,0 +1,114 @@
+%%==============================================================================
+%% Compiler diagnostics
+%%==============================================================================
+
+-module(els_sheldon_diagnostics).
+
+%%==============================================================================
+%% Behaviours
+%%==============================================================================
+
+-behaviour(els_diagnostics).
+
+%%==============================================================================
+%% Exports
+%%==============================================================================
+
+-export([ is_default/0
+        , run/1
+        , source/0
+        ]).
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+
+-include("els_lsp.hrl").
+-include_lib("kernel/include/logger.hrl").
+
+%%==============================================================================
+%% Callback Functions
+%%==============================================================================
+
+-spec is_default() -> boolean().
+is_default() ->
+  false.
+
+-spec run(uri()) -> [els_diagnostics:diagnostic()].
+run(Uri) ->
+  case els_utils:project_relative(Uri) of
+    {error, not_relative} -> [];
+    RelFile ->
+      try
+        RegEx = "[_@./#&+-=*]",
+        Mod = rebar3_sheldon_ast,
```

Comment:
Because we don't run `sheldon` together with the project but only when spellcheck diagnostics is running since it scares the `dialyzer` - I decided to use this approach.

Reply:
Fixed.